spring-pulsar-docs/src/main/asciidoc/pulsar.adoc

```diff
@@ -1222,6 +1262,116 @@ PulsarTopic partitionedTopic {
 ----
 ====
 
+[[pulsar-headers]]
+==== Pulsar Headers
+
+In this section, we will see how we can use the message metadata that Pulsar provides in its Message as Pulsar headers in a Spring application.
+First, let's examine all the available Pulsar headers from the message metadata.
+
+.[.underline]#Aailable **Pulsar Message Metadata as Spring Headers**#
+[%collapsible]
+====
+https://github.com/spring-projects-experimental/spring-pulsar/blob/main/spring-pulsar/src/main/java/org/springframework/pulsar/support/PulsarHeaders.java[Click here] to see the available Pulsar headers.
+====
+
+===== Accessing Pulsar Headers in Single Record based Consumer
+
+Here is an example of how you can access the various Pulsar Headers in an application that is using single record mode of consuming.
+
+====
+[source,java]
+----
+@PulsarListener(topics = "simpleListenerWithHeaders")
+void simpleListenerWithHeaders(String data, @Header(PulsarHeaders.MESSAGE_ID) MessageId messageId,
+                @Header(PulsarHeaders.RAW_DATA) byte[] rawData,
+                @Header("foo") String foo) {
+
+}
+----
+====
+
+In this example, we are consuming the data as `String`.
+In addition, we are also extracting the message meatadata as Spring headers using the `Header` annotation from the Spring Framework.
+We just randomly picked the headers for message id and raw data.
+You can also extract custom message properties set when publishing using the `Header` annotation as observed in the retrieval of the property `foo` in the above example code.
+
+Now let us try to receive this same data, but with the envelope of a Pulsar `Message`.
```

Comment:
I would be in favor in reducing the code samples from 6 down to 2. Here is my reasoning:

The method of accessing the headers is the same in all cases (via the `@Header` annotation) w/ only 2 interesting facts:
1. When accessing in batch mode (aka N records) then there is a. parallel list of N header entries.
2. When accessing data as pulsar msg then you have the option of directly accessing the message metadata from the pulsar message

We could probably get by w/ 1 example and just talk to the points or at most 2 would cover the interesting pieces. This will simplify the maintenance of this doc and make it less likely to get bit rot. 

Wdyt?

Reply:
This is great feedback. Let's do this.